File: Leetcode/PY/Leetcode554-砖墙.py
from typing import List
from collections import defaultdict
class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        hash = defaultdict(int)
        for line in wall:
            s = 0
            for i in range(0,len(line) - 1):
                s += line[i]
                hash[s] += 1

        res = 0
        for _,v in hash.items():
            res = max(res,v)

        return len(wall) - res

File: Leetcode/Rust/Leetcode554-砖墙.rs
use std::collections::HashMap;
impl Solution {
    pub fn least_bricks(wall: Vec<Vec<i32>>) -> i32 {
        let mut hash = HashMap::new();
        for line in wall.iter(){
            let mut sum = 0;
            let end = line.len() - 1;
            for i in 0..end{
               sum += line[i]; 
               let count = hash.entry(sum).or_insert(0);
                *count += 1;
            }
        }

        let mut res = 0;
        for (k,v) in hash{
            res = res.max(v);
        }
        wall.len() as i32 - res
    }
}